### DISCUSSION TITLE
File upload order is random

### DISCUSSION THREAD

**Discussion by SeanDylanGoff** [directus/directus#16170] [Feature Requests]
When uploading multiple files through POST /files, the order of the returned file Objects is random.
I think the upload order should be preserved, so the user doesn't have to resort to hacks to determine which file is which (or upload them separately)

### To Reproduce

```const f = new FormData();
f.append("title", "file1");
f.append("data", new Blob(["1"]));
f.append("title", "file2");
f.append("data", new Blob(["22"]));
fetch('https://[TESTDOMAIN].directus.app/files', {
    method: 'POST',
    body: f,
    headers: {
        authorization: 'Bearer [APITOKEN]
    }
}).then(res=>res.json().then(({data})=>console.log(data.map(f=>f.title))))
```

**Comment by rijkvanzanten** [comment]
The default sort order (for files or other wise) is the primary key, so it's not entirely random 🙂 

That being said, I agree it would be a nice improvement to keep the output ordered the same as the input, so lets consider that a feature request 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
